testsuite/features/step_definitions/retail_steps.rb

```diff
@@ -42,15 +42,129 @@
+  conf2 = 'GATEWAYDEV=eth0'
+  file2 = '/etc/sysconfig/network'
+  nodes.each do |node|
+    next if node.nil?
+
+    domain, _code = node.run('grep \'^search\' /etc/resolv.conf | sed \'s/^search//\'')
+    conf = "DOMAIN='#{domain.strip}'\\nDEVICE='eth1'\\nSTARTMODE='auto'\\nBOOTPROTO='dhcp'\\nDNS1='#{proxy}'"
+    service =
+      if node.os_family =~ /^rocky/
+        'NetworkManager'
+      else
+        'network'
+      end
+    node.run("echo -e \"#{conf}\" > #{file} && echo -e \"#{conf2}\" > #{file2} && systemctl restart #{service}")
+  end
+  # /etc/netplan/01-netcfg.yaml
+  nodes = [get_target('deblike_minion')]
+  source = "#{File.dirname(__FILE__)}/../upload_files/01-netcfg.yaml"
+  dest = '/etc/netplan/01-netcfg.yaml'
+  nodes.each do |node|
+    next if node.nil?
+
+    return_code = file_inject(node, source, dest)
+    raise ScriptError, 'File injection failed' unless return_code.zero?
+
+    node.run('netplan apply')
+  end
+  # PXE boot minion
+  if $pxeboot_mac
+    step 'I restart the network on the PXE boot minion'
+  end
+end
+
 Then(/^"([^"]*)" should communicate with the server using public interface$/) do |host|
   node = get_target(host)
-  _result, return_code = node.run("ping -c 1 -I #{node.public_interface} #{get_target('server').public_ip}", check_errors: false)
+  _result, return_code = node.run("ping -n -c 1 -I #{node.public_interface} #{get_target('server').public_ip}", check_errors: false)
   unless return_code.zero?
     sleep 2
     puts 're-try ping'
-    node.run("ping -c 1 -I #{node.public_interface} #{get_target('server').public_ip}")
+    node.run("ping -n -c 1 -I #{node.public_interface} #{get_target('server').public_ip}")
+  end
+  get_target('server').run("ping -n -c 1 #{node.public_ip}")
+end
+
+When(/^I activate the private network on the proxy$/) do
+  node = get_target('proxy')
+  _result, return_code = node.run('which nmcli')
+  if return_code.zero?
+    # Network manager: we give eth1 precedence over eth0
+    #                  otherwise the name server we get from DHCP is lost at the end of the list
+    #                  (the name servers list in resolv.conf is limited to 3 entries)
+    cmd = 'nmcli connection modify "Wired connection 1" ipv4.dns-priority 20 && ' \
+          'nmcli device modify eth0 ipv4.dns-priority 20 && ' \
+          'nmcli connection modify "Wired connection 2" ipv4.dns-priority 10 && ' \
+          'nmcli device modify eth1 ipv4.dns-priority 10'
+  else
+    # Wicked: is there a way to give eth1 precedence?
+    #         we use a static setting for the name server instead
+    static_dns = net_prefix + PRIVATE_ADDRESSES['dhcp_dns']
+    cmd = 'echo -e "BOOTPROTO=dhcp\nSTARTMODE=auto\n" > /etc/sysconfig/network/ifcfg-eth1 && ' \
+          'ifup eth1 && ' \
+          "sed -i 's/^NETCONFIG_DNS_STATIC_SERVERS=\".*\"/NETCONFIG_DNS_STATIC_SERVERS=\"#{static_dns}\"/' /etc/sysconfig/network/config && " \
+          'netconfig update -f'
+  end
+  node.run(cmd)
```

Comment:
If the proxy runs on a container, all this will be applied inside the container instead of the proxy host. That's what we aim to do here?

Reply:
Gasp... It's a copy and paste from the same step applied to terminals, and terminals don't use containers...

I'm wondering why I did not see problems with that when I tested with Uyuni podman. Uyuni podman is based on opensuse 15.5, therefore uses wicked, but is containerized. So this code will have run inside the container. Maybe that just works? :thinking: 

In any case `run_local` seems safer, yes.